GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs

```diff
@@ -1176,5 +1179,57 @@ internal override FightData.EncounterMode GetEncounterMode(CombatData combatData
             }
             return FightData.EncounterMode.Normal;
         }
+
+        protected override void SetInstanceBuffs(ParsedEvtcLog log)
+        {
+            base.SetInstanceBuffs(log);
+            IReadOnlyList<AbstractBuffEvent> voidwalker = log.CombatData.GetBuffData(AchievementEligibilityVoidwalker);
+
+            // Added a CM mode check because the eligibility has been bugged for a while and showed up in normal mode.
+            bool isCM = GetEncounterMode(log.CombatData, log.AgentData, log.FightData) == FightData.EncounterMode.CM;
+            bool hasVoidwalkerBeenAdded = false;
+
+            if (log.FightData.Success)
+            {
+                if (voidwalker.Any() && isCM)
+                {
+                    foreach (Player p in log.PlayerList)
+                    {
+                        if (p.HasBuff(log, AchievementEligibilityVoidwalker, log.FightData.FightEnd - ServerDelayConstant))
+                        {
+                            InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityVoidwalker], 1));
+                            hasVoidwalkerBeenAdded = true;
+                            break;
+                        }
+                    }
+                }
+                if (!hasVoidwalkerBeenAdded && CustomCheckVoidwalkerEligibility(log) && isCM) // In case all 10 players already have voidwalker
+                {
+                    InstanceBuffs.Add((log.Buffs.BuffsByIds[AchievementEligibilityVoidwalker], 1));
+                }
+            }
+        }
+
+        private static bool CustomCheckVoidwalkerEligibility(ParsedEvtcLog log)
+        {
+            IReadOnlyList<AgentItem> orbs = log.AgentData.GetNPCsByID((int)ArcDPSEnums.TrashID.PushableVoidAmalgamate);
+
+            foreach (AgentItem orb in orbs)
+            {
+                IReadOnlyDictionary<long, BuffsGraphModel> bgms = log.FindActor(orb).GetBuffGraphs(log);
+                if (bgms != null && bgms.ContainsKey(VoidEmpowerment))
```

Comment:
ContainsKey(VoidEmpowerment) to be replaced with TryGetValue(VoidEmpowerment, out BuffsGraphModel bgm), remove the one below